### PR TITLE
Add customer popup front end

### PR DESCRIPTION
**Pull request by Jamalat-shamallakh**
### Related:
related to #158

### CODE REVIEW

File: client/src/Components/Layouts/Customers/style.css [hunk 1]
```diff
@@ -43,4 +43,75 @@
...
+}
+
+.ant-modal-footer {
+    text-align: left !important;
+    direction: ltr !important;
+}
+
+.nameconatainer {
+    display: inline-block;
+    margin-top: 15px;
```
Comment (hshahwan) [inline client/src/Components/Layouts/Customers/style.css]: change to rem

File: client/src/Components/Layouts/Customers/index.jsx [hunk 2]
```diff
@@ -92,35 +92,90 @@ export default class Customers extends Component {
...
+                            <CollectionCreateForm
+                                wrappedComponentRef={this.saveFormRef}
+                                visible={this.state.visible}
+                                onCancel={this.handleCancel}
+                                onCreate={this.handleCreate}
+                                onchange={this.onchange}
+                            />
+                            <div className="filtercontainer">
+                                <div classNam="filtercontainer__orderdate">
+                                    <RangePicker
```
Comment (hshahwan) [inline client/src/Components/Layouts/Customers/index.jsx]: delete the footer of RangePicker by using
``` 
  .ant-calendar-footer-btn{
      display: none;
  }
```

File: client/src/Components/Layouts/Customers/addcustomer.jsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                })(<Select>
+                  <Option value="true">فعال</Option>
+                  <Option value="false">غير فعال</Option>
+                </Select>)}
+              </Form.Item>
+            </div>
+            <div className='modalform__left-container'>
+              <Form.Item label="البريدالالكتروني" layout="horizontal" className='modalform_formitem'>
+                {getFieldDecorator('email', {
+                  rules: [{ required: true, message: '', validator: this.handelEmail }],
```
Comment (hshahwan) [inline client/src/Components/Layouts/Customers/addcustomer.jsx]: add input for passowrd.
```
  {getFieldDecorator("password", {
                  rules: [
                    {
                      message: "الرجاء ملئ الحقل",
                      required: true
                    }
                  ]
                })(<Input type="password" id="password" />)}
```

File: client/src/Components/Layouts/Customers/index.jsx [hunk 4]
```diff
@@ -92,35 +92,90 @@ export default class Customers extends Component {
...
-                            <div classNam="filtercontainer__orderdate">
-                                <RangePicker
-                                    showTime={{ format: 'HH:mm' }}
-                                    format="YYYY-MM-DD HH:mm"
-                                    placeholder={['من', 'الى']}
-                                    onChange={e => this.filterfunction(e, '', 'date')}
-                                   className="containercustomers__customer-rangpicker"
-                                />
-                                <span className='filtercontainer__orderdate-date'>فلترة حسب الوقت</span>
+        if (this.state.customers) {
```
Comment (hshahwan) [inline client/src/Components/Layouts/Customers/index.jsx]: if there is no data in the state this message will appear
```
Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.
```

to solve it add else statment and create a loading page

File: client/src/Components/Layouts/Customers/addcustomer.jsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        <Modal
+          visible={visible}
+          title={(
+            <div>
+              <Icon type="user-add" className="modeltitle" />
+              إضافة مستخدم
+              {' '}
+            </div>
+          )}
+          okText="أنشئ"
```
Comment (hshahwan) [inline client/src/Components/Layouts/Customers/addcustomer.jsx]: change the text to  اضافة

File: client/src/Components/Layouts/Customers/addcustomer.jsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+            <div className='modalform__right-container'>
+              <Form.Item label="الاسم" layout="horizontal" className='modalform_formitem'>
+                {getFieldDecorator('name', {
+                  rules: [{ required: true, validator: this.handelName }],
+                })(<div>
+                  <Input />
+                </div>
+                )}
+              </Form.Item>
+              <div style={{ display: 'flex' }}>
```
Comment (hshahwan) [inline client/src/Components/Layouts/Customers/addcustomer.jsx]: delete inline style

File: client/src/Components/Layouts/Customers/addcustomer.jsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+                <Form.Item label="الهاتف" layout="horizontal" className='modalform_formitem'>
+                  {getFieldDecorator('phone', {
+                    rules: [
+                      {
+                        required: true,
+                        validator: this.handlePhone
+
+                      }],
```
Comment (engshorouq) [inline client/src/Components/Layouts/Customers/addcustomer.jsx]: you can make it like that :
```
rules: [
                    {
                      required: true,
                      message: "رقم الهاتف يجب أن يتكون من  تسعة أرقام",
                      pattern: /^[0-9]{9}$/
                    }
                  ]
```

File: client/src/Components/Layouts/Customers/index.jsx [hunk 8]
```diff
@@ -92,35 +92,90 @@ export default class Customers extends Component {
...
+                method: 'POST',
+                headers: { 'content-type': 'application/json' },
+                body: JSON.stringify(addCustomer)
+            }).then(res => res.json())
+                .then(result => {
+                    if (result.result) {
+                        let newcustomer = [...this.state.customers];
+                        newcustomer.push(result.result[0])
+                        let newallData = [...this.state.customers];
+                        newallData.push(result.result[0])
```
Comment (engshorouq) [inline client/src/Components/Layouts/Customers/index.jsx]: why repeat the same code you can make it like that : 
```
this.setState(prev => {
return { customers: [...prev.customers, result.result[0]], allData: [...prev.allData, result.result[0]]}
})
```

File: client/src/Components/Layouts/Customers/index.jsx [hunk 9]
```diff
@@ -92,35 +92,90 @@ export default class Customers extends Component {
...
+                        this.setState({
+                            customers: newcustomer,
+                            allData: newallData
+                        })
+                    }
+                })
+            form.resetFields();
+            this.setState({ visible: false });
```
Comment (engshorouq) [inline client/src/Components/Layouts/Customers/index.jsx]: this line is repeated more than the one you can put it in one function and use it : 
```
visibilityHandle = () => {
this.setState(prev => {
return {visible: !prev.visible};
})
}
```
and call it in `handleCancel, showModal` and here

File: client/src/App.jsx [hunk 10]
```diff
@@ -1,7 +1,8 @@
 import React, { Component } from "react";
 import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
-import { Icon } from 'antd'
-import Customers from './Components/Layouts/Customers'
+import {Icon}from 'antd';
+import Customers from './Components/Layouts/Customers/index';
+import CollectionsPage from './Components/Layouts/Customers/deletecustomer'
```
Comment (hshahwan) [inline client/src/App.jsx]: delete this import

**Pull request merged**